ledger/participant-integration-api/src/main/scala/platform/store/utils/ConcurrencyLimiter.scala

```diff
@@ -14,20 +14,27 @@ class NoConcurrencyLimiter extends ConcurrencyLimiter {
   override def execute[T](task: => Future[T]): Future[T] = task
 }
 
+/** @param executionContext - needed to run limiter's internal book-keeping. Does not run the submitted tasks.
+  * @param parentO - used to define a hierarchy (a tree) of concurrency limiters such that we can
+  *                  express constraints like: for 5 local limiters each running at most 3 concurrent tasks
+  *                  set a global limit of at most 10 concurrent in total.
+  */
+// TODO etq: Separate 1) concurrency limiting from 2) composition of limiters
```

Comment:
Isn't this is the same as to have two original ConcurrencyLimiter used so: child.execute(parent.execute(stuff))?
If yes, I would suggest not changing this file.